src/rgw/rgw_admin.cc

```diff
@@ -4234,6 +4237,42 @@ int main(int argc, char **argv)
     }
   }
 
+  if (opt_cmd == OPT_BUCKETS_LIMIT_CHECK) {
+
+	  void *handle;
+      metadata_key = "user";
+	  int max = 1000;
+	  int ret = store->meta_mgr->list_keys_init(metadata_key, &handle);
+	  if (ret < 0) {
+		  cerr << "ERROR: buckets limit check can't get user metadata_key: "
+			   << cpp_strerror(-ret) << std::endl;
+		  return -ret;
+	  }
+
+	  bool truncated;
+
+	  /* list users in groups of max-keys, then perform user-bucket
+	   * limit-check on each group */
+	  do {
+		  std::list<std::string> user_ids;
+		  ret = store->meta_mgr->list_keys_next(handle, max, user_ids,
```

Comment:
@mattbenjamin maybe if uid is passed in the command line, just show stats of a specific user?